public/javascripts/janus.js

```diff
@@ -279,4 +279,30 @@ jQuery(function($){
         }
     });
 
+    // auto-logout (preference persisted with local storage)
+    $("#auto_logout_switch").each(function(_, autoLogoutSwitchElement){
+        const LOCAL_STORAGE_KEY__AUTO_LOGOUT = "autoLogout"
+        autoLogoutSwitchElement.checked = localStorage.getItem(LOCAL_STORAGE_KEY__AUTO_LOGOUT) === "true";
+        autoLogoutSwitchElement.onchange = (event) => {
+            autoLogoutSwitchElement.checked = event.target.checked;
+            localStorage.setItem(LOCAL_STORAGE_KEY__AUTO_LOGOUT, event.target.checked.toString());
+        };
+
+        $("a[href*='/console?permissionId=']").each(function(_, el){
+            el.onclick = (clickEvent) => {
+                if(autoLogoutSwitchElement.checked) {
+                    clickEvent.preventDefault();
+                    const targetHref = el.href;
+                    console.log("Silently attempting logout before navigating to", targetHref)
+                    fetch("https://signin.aws.amazon.com/logout", {
```

Comment:
Should we put a timeout on this, so that we can be pretty sure the user's intent will be followed at some point?
I worry network issues or unusual browser settings (now or in the future) could leave Janus broken if they mean this call doesn't complete?

Reply:
great point, fixed in https://github.com/guardian/janus-app/pull/439/files/065b979f76e173b8a555342d726aa87e17e40910..c8a34438bb9739b61fbe69955b99443ce9fecf69